pkg/agents/team6/treaties.go

```diff
@@ -0,0 +1,375 @@
+					return a.considerTreatyUsingUtility(t)
+				} else {
+					if t.ConditionValue() >= levels.strongLevel {
+						return true
+					} else {
+						a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+						return a.considerTreatyUsingUtility(t)
+
+					}
+				}
+			case "Narcissist":
+				a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+				return a.considerTreatyUsingUtility(t)
+			default:
+				a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+				return a.considerTreatyUsingUtility(t)
+			}
+		// Floor
+		case 2:
+			return a.considerTreatyUsingUtility(t)
+		// AvailableFood
+		case 3:
+			switch a.currBehaviour.string() {
+			case "Altruist":
+				return true
+			case "Collectivist":
+				if t.ConditionOp() == messages.LE || t.ConditionOp() == messages.LT {
+					a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+					return a.considerTreatyUsingUtility(t)
+				} else {
+					if a.convertToTakeFoodAmount(float64(t.ConditionValue()), t.Request(), t.RequestValue()) <= 2 { // 2 is the amount needed to y sufficient to go from critical to WeakLevel
+						a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+						return a.considerTreatyUsingUtility(t)
+					} else {
+						return true
+					}
+				}
+			case "Selfish":
+				if t.ConditionOp() == messages.LE || t.ConditionOp() == messages.LT {
+					a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+					return a.considerTreatyUsingUtility(t)
+				} else {
+					if a.convertToTakeFoodAmount(float64(t.ConditionValue()), t.Request(), t.RequestValue()) <= 60 { // change to a.HealthInfo().MaxFoodIntake
+						a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+						return a.considerTreatyUsingUtility(t)
+					} else {
+						return true
+					}
+				}
+
+			case "Narcissist":
+				a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+				return a.considerTreatyUsingUtility(t)
+			default:
+				a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+				return a.considerTreatyUsingUtility(t)
+			}
+
+		default:
+			a.Log("I used considerTreatyUsingUtility", infra.Fields{"my floor:": a.Floor(), "my social motive": a.currBehaviour.string()})
+			return a.considerTreatyUsingUtility(t)
+		}
+	} else {
+		return false
+	}
+}
```

Comment:
Function is too long - use functions to shorten it

Reply:
It is like a lookup table, putting the individual switch cases into the function is not going to improve readability

Reply:
No, this is extremely difficult to read. 
The way you handle `Collectivist`, and `Selfish` should be in functions.
The way you handle `HP` and `AvailableFood` should be in functions.

Reply:
This is largely resolved by #289